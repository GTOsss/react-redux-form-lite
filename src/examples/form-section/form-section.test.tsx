import React from 'react';
import {mount} from 'enzyme';
import {Provider} from 'react-redux';
import {createStore, applyMiddleware, combineReducers} from 'redux';
import ReduxThunkTester from 'redux-thunk-tester';
import getFormSection from './form-section';
import {reducer} from '../../index';

interface IValues {
  firstName?: string;
  lastName?: string;
  email?: string;
  sex?: string;
  favoriteColor?: string;
  employed?: string;
  notes?: string;
}

const renderComponent = (onSubmit?) => {
  const reduxThunkTester = new ReduxThunkTester();
  const FormSection = getFormSection();

  const store = createStore(
    combineReducers({reduxForm: reducer}),
    applyMiddleware(reduxThunkTester.createReduxThunkHistoryMiddleware()),
  );

  const component = mount(
    <Provider store={store}>
      <FormSection onSubmit={onSubmit} />
    </Provider>,
  );

  return {
    reduxThunkTester,
    getFormState: () => store.getState().reduxForm.simple as IReduxFormState<IValues>,
    component,
  };
};

describe('<FormSection />', () => {
  test('Render simple form (inputs).', () => {
    const {component} = renderComponent();
    expect(component.find('input')).toMatchSnapshot();
  });

  test('Render simple form after push user (inputs).', () => {
    const {component} = renderComponent();
    component.find('#addUser').simulate('click');
    expect(component.find('input')).toMatchSnapshot();
  });

  test('Register form and fields: store.', () => {
    const {getFormState} = renderComponent();
    expect(getFormState()).toMatchSnapshot();
  });

  test('Render simple form after push user (store).', () => {
    const {component, getFormState} = renderComponent();
    component.find('#addUser').simulate('click');
    expect(getFormState()).toMatchSnapshot();
  });

  test('Focus field: store', () => {
    const {getFormState, component} = renderComponent();
    component.find('input').at(0).simulate('focus');
    expect(getFormState()).toMatchSnapshot();
  });

  test('Changed field: store', () => {
    const {component, getFormState} = renderComponent();

    component.find('input').at(0).simulate('focus');
    component.find('input').at(0).simulate('change', {target: {value: 't'}});
    component.find('input').at(0).simulate('change', {target: {value: 'te'}});
    component.find('input').at(0).simulate('change', {target: {value: 'tes'}});
    component.find('input').at(0).simulate('change', {target: {value: 'test'}});

    expect(getFormState()).toMatchSnapshot();
  });

  test('Blur field: store', () => {
    const {component, getFormState} = renderComponent();

    component.find('input').at(0).simulate('focus');
    component.find('input').at(0).simulate('change', {target: {value: 't'}});
    component.find('input').at(0).simulate('change', {target: {value: 'te'}});
    component.find('input').at(0).simulate('change', {target: {value: 'tes'}});
    component.find('input').at(0).simulate('change', {target: {value: 'test'}});
    component.find('input').at(0).simulate('blur');

    expect(getFormState()).toMatchSnapshot();
  });

  test('onSubmit', () => {
    const onSubmit = jest.fn();
    const {component} = renderComponent(onSubmit);

    component.find('form').simulate('submit');
    expect(onSubmit.mock.calls.length).toBe(1);
    expect(onSubmit.mock.calls).toMatchSnapshot();
  });
});
