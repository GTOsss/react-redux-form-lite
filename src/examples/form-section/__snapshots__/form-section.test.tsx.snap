// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormSection /> Blur field: store 1`] = `
Object {
  "form": Object {
    "activeField": "",
    "blurred": true,
    "changed": true,
    "errorsMap": Object {},
    "focused": true,
    "hasErrors": false,
    "hasWarnings": false,
    "submitted": false,
    "warningsMap": Object {},
  },
  "meta": Object {
    "userFormSection": Object {
      "contacts": Object {
        "email": Object {
          "active": false,
          "blurred": false,
          "changed": false,
          "error": "",
          "focused": false,
          "warning": "",
        },
      },
      "firstName": Object {
        "active": false,
        "blurred": true,
        "changed": true,
        "error": "",
        "focused": true,
        "warning": "",
      },
      "lastName": Object {
        "active": false,
        "blurred": false,
        "changed": false,
        "error": "",
        "focused": false,
        "warning": "",
      },
    },
  },
  "values": Object {
    "userFormSection": Object {
      "contacts": Object {
        "email": undefined,
      },
      "firstName": "test",
      "lastName": undefined,
    },
    "users": Array [],
  },
}
`;

exports[`<FormSection /> Changed field: store 1`] = `
Object {
  "form": Object {
    "activeField": "userFormSection.firstName",
    "blurred": false,
    "changed": true,
    "errorsMap": Object {},
    "focused": true,
    "hasErrors": false,
    "hasWarnings": false,
    "submitted": false,
    "warningsMap": Object {},
  },
  "meta": Object {
    "userFormSection": Object {
      "contacts": Object {
        "email": Object {
          "active": false,
          "blurred": false,
          "changed": false,
          "error": "",
          "focused": false,
          "warning": "",
        },
      },
      "firstName": Object {
        "active": true,
        "blurred": false,
        "changed": true,
        "error": "",
        "focused": true,
        "warning": "",
      },
      "lastName": Object {
        "active": false,
        "blurred": false,
        "changed": false,
        "error": "",
        "focused": false,
        "warning": "",
      },
    },
  },
  "values": Object {
    "userFormSection": Object {
      "contacts": Object {
        "email": undefined,
      },
      "firstName": "test",
      "lastName": undefined,
    },
    "users": Array [],
  },
}
`;

exports[`<FormSection /> Focus field: store 1`] = `
Object {
  "form": Object {
    "activeField": "userFormSection.firstName",
    "blurred": false,
    "changed": false,
    "errorsMap": Object {},
    "focused": true,
    "hasErrors": false,
    "hasWarnings": false,
    "submitted": false,
    "warningsMap": Object {},
  },
  "meta": Object {
    "userFormSection": Object {
      "contacts": Object {
        "email": Object {
          "active": false,
          "blurred": false,
          "changed": false,
          "error": "",
          "focused": false,
          "warning": "",
        },
      },
      "firstName": Object {
        "active": true,
        "blurred": false,
        "changed": false,
        "error": "",
        "focused": true,
        "warning": "",
      },
      "lastName": Object {
        "active": false,
        "blurred": false,
        "changed": false,
        "error": "",
        "focused": false,
        "warning": "",
      },
    },
  },
  "values": Object {
    "userFormSection": Object {
      "contacts": Object {
        "email": undefined,
      },
      "firstName": undefined,
      "lastName": undefined,
    },
    "users": Array [],
  },
}
`;

exports[`<FormSection /> Register form and fields: store. 1`] = `
Object {
  "form": Object {
    "activeField": "",
    "blurred": false,
    "changed": false,
    "errorsMap": Object {},
    "focused": false,
    "hasErrors": false,
    "hasWarnings": false,
    "submitted": false,
    "warningsMap": Object {},
  },
  "meta": Object {
    "userFormSection": Object {
      "contacts": Object {
        "email": Object {
          "active": false,
          "blurred": false,
          "changed": false,
          "error": "",
          "focused": false,
          "warning": "",
        },
      },
      "firstName": Object {
        "active": false,
        "blurred": false,
        "changed": false,
        "error": "",
        "focused": false,
        "warning": "",
      },
      "lastName": Object {
        "active": false,
        "blurred": false,
        "changed": false,
        "error": "",
        "focused": false,
        "warning": "",
      },
    },
  },
  "values": Object {
    "userFormSection": Object {
      "contacts": Object {
        "email": undefined,
      },
      "firstName": undefined,
      "lastName": undefined,
    },
    "users": Array [],
  },
}
`;

exports[`<FormSection /> Render simple form (inputs). 1`] = `
Array [
  <input
    name="userFormSection.firstName"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="First Name"
    type="text"
    value=""
  />,
  <input
    name="userFormSection.lastName"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="Last Name"
    type="text"
    value=""
  />,
  <input
    name="userFormSection.contacts.email"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="Email"
    type="email"
    value=""
  />,
]
`;

exports[`<FormSection /> Render simple form after push user (inputs). 1`] = `
Array [
  <input
    name="userFormSection.firstName"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="First Name"
    type="text"
    value=""
  />,
  <input
    name="userFormSection.lastName"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="Last Name"
    type="text"
    value=""
  />,
  <input
    name="userFormSection.contacts.email"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="Email"
    type="email"
    value=""
  />,
  <input
    name="userFormSection.users[0].firstName"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    type="text"
    value=""
  />,
]
`;

exports[`<FormSection /> Render simple form after push user (store). 1`] = `
Object {
  "form": Object {
    "activeField": "",
    "blurred": false,
    "changed": false,
    "errorsMap": Object {},
    "focused": false,
    "hasErrors": false,
    "hasWarnings": false,
    "submitted": false,
    "warningsMap": Object {},
  },
  "meta": Object {
    "userFormSection": Object {
      "contacts": Object {
        "email": Object {
          "active": false,
          "blurred": false,
          "changed": false,
          "error": "",
          "focused": false,
          "warning": "",
        },
      },
      "firstName": Object {
        "active": false,
        "blurred": false,
        "changed": false,
        "error": "",
        "focused": false,
        "warning": "",
      },
      "lastName": Object {
        "active": false,
        "blurred": false,
        "changed": false,
        "error": "",
        "focused": false,
        "warning": "",
      },
      "users": Array [
        Object {
          "firstName": Object {
            "active": false,
            "blurred": false,
            "changed": false,
            "error": "",
            "focused": false,
            "warning": "",
          },
        },
      ],
    },
  },
  "values": Object {
    "userFormSection": Object {
      "contacts": Object {
        "email": undefined,
      },
      "firstName": undefined,
      "lastName": undefined,
      "users": Array [
        Object {
          "firstName": undefined,
        },
      ],
    },
    "users": Array [
      Object {
        "id": 0,
      },
    ],
  },
}
`;

exports[`<FormSection /> onSubmit 1`] = `
Array [
  Array [
    Object {
      "actions": Object {
        "arrayPush": [Function],
        "arrayRemove": [Function],
        "blur": [Function],
        "change": [Function],
        "changeSubmitted": [Function],
        "focus": [Function],
        "registerField": [Function],
        "registerForm": [Function],
        "removeField": [Function],
        "removeForm": [Function],
        "resetForm": [Function],
        "setInitialValues": [Function],
        "updateFormState": [Function],
        "updateValidateAndWarningMessages": [Function],
        "updateValidateMessage": [Function],
        "updateValidateMessages": [Function],
        "updateWarningMessage": [Function],
        "updateWarningMessages": [Function],
      },
      "state": Object {
        "form": Object {
          "activeField": "",
          "blurred": false,
          "changed": false,
          "errorsMap": Object {
            "userFormSection": Object {
              "firstName": "Field required.",
            },
          },
          "focused": false,
          "hasErrors": true,
          "hasWarnings": false,
          "submitted": true,
          "warningsMap": Object {},
        },
        "meta": Object {
          "userFormSection": Object {
            "contacts": Object {
              "email": Object {
                "active": false,
                "blurred": false,
                "changed": false,
                "error": "",
                "focused": false,
                "warning": "",
              },
            },
            "firstName": Object {
              "active": false,
              "blurred": false,
              "changed": false,
              "error": "Field required.",
              "focused": false,
              "warning": "",
            },
            "lastName": Object {
              "active": false,
              "blurred": false,
              "changed": false,
              "error": "",
              "focused": false,
              "warning": "",
            },
          },
        },
        "values": Object {
          "userFormSection": Object {
            "contacts": Object {
              "email": undefined,
            },
            "firstName": undefined,
            "lastName": undefined,
          },
          "users": Array [],
        },
      },
      "values": Object {
        "userFormSection": Object {
          "contacts": Object {
            "email": undefined,
          },
          "firstName": undefined,
          "lastName": undefined,
        },
        "users": Array [],
      },
      "wizard": undefined,
    },
  ],
]
`;
